### PR TITLE
Merge 

### PR DESCRIPTION
**Pull request by adeelq**

### CODE REVIEW

File: code/Dotdigitalgroup/Email/controllers/EmailController.php [hunk 1]
```diff
@@ -295,6 +295,7 @@ public function accountcallbackAction()
     {
         $params = $this->getRequest()->getParams();
         $helper = Mage::helper('ddg');
+        $error = true;
```
Comment (cdiacon) [inline code/Dotdigitalgroup/Email/controllers/EmailController.php]: @adeelq just not sure if the default the error should be true?!?

if the error is false and the error occur it's easy to find where the things go wrong(becomes true).

Comment (adeelq) [inline code/Dotdigitalgroup/Email/controllers/EmailController.php]: I had another code but thought this way would be short code..but yeah you right better to separate the errors

File: code/Dotdigitalgroup/Email/controllers/EmailController.php [hunk 2]
```diff
@@ -306,13 +306,18 @@ public function accountcallbackAction()
                     $helper->saveApiEndPoint($params['apiEndpoint']);
                 }
                 if ($apiConfigStatus && $dataFieldsStatus && $addressBookStatus && $syncStatus) {
-                    $error = false;
                     $this->sendAjaxResponse(false, $this->_getSuccessHtml());
+                } else {
+                    $error = true;
```
Comment (cdiacon) [inline code/Dotdigitalgroup/Email/controllers/EmailController.php]: @adeelq what is the callback error message, do we display something that went wrong?

Comment (adeelq) [inline code/Dotdigitalgroup/Email/controllers/EmailController.php]: just general message saying something went wrong contact customer service. we do not tell anything specific.

Comment (cdiacon) [inline code/Dotdigitalgroup/Email/controllers/EmailController.php]: so this is just a simple refactor with no error message?

Comment (adeelq) [inline code/Dotdigitalgroup/Email/controllers/EmailController.php]: no error message goes now. before i was sending $this->sendAjaxResponse(true, 'Error'); now if error occurs always send $this->sendAjaxResponse(true, $this->_getErrorHtml());

Comment (cdiacon) [inline code/Dotdigitalgroup/Email/controllers/EmailController.php]: _Sorry,_ something went wrong whilst trying to create your new dotmailer account_
the account may be created successfully, can be something wrong with mapping.

Error makes more sense then ^

Comment (adeelq) [inline code/Dotdigitalgroup/Email/controllers/EmailController.php]: sure it does make sense but i got only one error message from Ben to show. Here in code with can show error message according to each scenario. Should we be doing that

Comment (cdiacon) [inline code/Dotdigitalgroup/Email/controllers/EmailController.php]: so if this is not set - it's an **error** : 

```
if (!empty($params['accountId']) && !empty($params['apiUser']) && !empty($params['pass']) && !empty($params['secret'])) {
```
everything else should be a **warnning** with possible message to display

Comment (adeelq) [inline code/Dotdigitalgroup/Email/controllers/EmailController.php]: yup also other checks too

**Pull request merged**